### ISSUE TITLE
Cannot save the trained ckpt file.

### ISSUE THREAD

**Issue by YutingLin** [smallcorgi/Faster-RCNN_TF#139]
Hi, 

I am trying to train the VOC2007 dataset for test. But found that the ckpt file cannot be saved.
I am running this test through docker container and trying to save the ckpt to the bind mounted path on the host machine(Ubuntu). 

The code seems working fine, without any error message. 
And the log also seems showing no specific problems.
> Wrote snapshot to: /data/notebook/faster-rcnn/output/test3/VGGnet_fast_rcnn_iter_100.ckpt
> done solving

But only the ckpt file cannot be saved. The other four files can be found in the same output directory.
(iter was set as 100 just for test)
![image](https://cloud.githubusercontent.com/assets/12992932/25573643/5dd10dd4-2e82-11e7-9257-28414c61e295.png)

At first, I thought it might caused by the conflict between host ownership and docker container ownership.
So I checked the ownership of docker container which is consistent with host machine root ownership.
![image](https://cloud.githubusercontent.com/assets/12992932/25573872/2198a956-2e84-11e7-8849-7292c7c817d6.png)

I cannot figure out what is the reason for causing this problem?
Do you have any suggestion for solving this problem?

Thank you very much.

**Comment by ghost** [comment]
第二个文件就是啊，

**Comment by ghost** [comment]
second  file is this

**Comment by LHagendoorn** [comment]
I don't know if the chinese guy answered your question, because, hey, he answered in chinese, but basically those files together form the ckpt file (or somehting like that) you can just adress the information as VGGnet_fast_rcnn_iter_100.ckpt

**Comment by YutingLin** [comment]
@ideaplat @TheLaurens Thanks mate. Such an easy and dumb question...

**Comment by chenjing8023** [comment]
Do you solve the problem? When we restore the model,just use VGGnet_fast_rcnn.cpkt.index?

**Comment by LHagendoorn** [comment]
Nope, just refer to it as  VGGnet_fast_rcnn.cpkt, that should just work, if not, maybe try using an absolute path?

**Comment by YutingLin** [comment]
@chenjing8023 Just as TheLaurens replied, VGGnet_fast_rcnn.cpkt works.( In my sample case, VGGnet_fast_rcnn_iter_100.cpkt)

**Comment by YutingLin** [comment]
The exported .meta/.index/.data-00000-of-00001 cannot be restored for test session, which repeatedly printing 'Waiting for VGGnet_fast_rcnn_iter_100.ckpt to exist...'. And the test session won't start.
I am not sure if this only happens on me.

After some research, I found out that you have to specify write_version=tf.train.SaverDef.V1 in tensorflow 0.12, which I am using. And the .ckpt file is saved and can be restored in the test session. 
You can add this args to /lib/fast_rcnn/train.py.

Reference:
https://www.tensorflow.org/api_docs/python/tf/train/Saver

**Comment by chenjing8023** [comment]
Yes,you are right . I have specify write_version=tf.train.SaverDef.V1,but I have this error when I restore the model in the test session. I don't know how to solve it.
cudaCheckError() failed : invalid resource handle

**Comment by YutingLin** [comment]
@chenjing8023 Although I am not sure your error is caused by CUDA version, but maybe you can check the CUDA version of your environment. For me, CUDA 8.0/cudnn 5 works fine.

**Comment by YutingLin** [comment]
@chenjing8023  Also, this post is similiar to your problem. #19

**Comment by LK1711** [comment]
@YutingLin could you please tell me what to what file should I add this line of code?

**Comment by YutingLin** [comment]
Hi,

Since it has been a while for my post, I have modified the code a little
bit.
Thus, I may be wrong with the accurate line where I changed.

Anyway, my  modification as follows:

lib/fast_rcnn/train.py

line 42:  self.saver = saver ⇒ self.saver = tf.train.Saver(max_to_keep=100,
write_version=tf.train.SaverDef.V1)

Actually, it should depend on which version of tensorflow you have
installed.


Best regards,

Lin


2018-03-15 17:09 GMT+09:00 LK1711 <notifications@github.com>:
